lib/bunny/session.rb

```diff
@@ -464,7 +464,7 @@ def blocked?
     # @param [String] uri amqp or amqps URI to parse
     # @return [Hash] Parsed URI as a hash
     def self.parse_uri(uri)
-      AMQ::Settings.parse_amqp_url(uri)
+      AMQ::Settings.configure(uri)
```

Comment:
❗️  Very important change that enables defaults merge for the parsed parameters.